frontend/src/redux/search/api.js

```diff
@@ -0,0 +1,15 @@
+import * as actions from "./actions";
+import api from "../../axios";
+
+export const search = (params, setData, setLoading) => {
+  return (dispatch) => {
+    dispatch(actions.searchAction());
+
+    api({sendToken: true})
+    .get("/search", {params: params})
+    .then((response) => {
+      setData(response.data)
+      setLoading(false)
+    })
+  };
+};
```

Comment:
Looks like we need a reducer here, you are setting the local state of the component with the given setstate parameters to the api function. If you need help on reducer part we might discuss

Reply:
It would be very helpful actually.